src/Codegen/Codegen.hs

```diff
@@ -521,11 +521,12 @@ genPrim I.After [time, lhs, rhs] _ = do
   (lhsVal , lhsStms ) <- genExpr lhs
   (rhsVal , rhsStms ) <- genExpr rhs
   Just later          <- return $ genLater $ extract lhs
+  let timeUnmarshalled  = unmarshal timeVal --is this a dangerous band aid?
```

Comment:
What is the issue you're worried about here?

Note that we also haven't actually worked out the types for time units quite yet. For now, I'm assuming all time values are some kind of integer value. If it's the fact that this only works if `timeVal` is some kind of integer value, then don't worry about that. You can assume it will be an integer, and that any kind of type error can and should be caught by the typechecker further upstream.

Reply:
Two stylistic nits: (1) if you ever have consecutive `let` definitions, you can keep them all in the same block, i.e.,
```haskell
let timeUnmarshalled = ...
    laterBlock = ...
```

But the other point is that you should just be able to perform the unmarshalling inline when `timeVal` is bound, using the `first` + `<$>` pattern we discussed before:

```haskell
(timeVal, timeStms) <- first unmarshal <$> genExpr time
```

Reply:
Thank you for your explanation! I was worried that the generated C code for later might not always need an unmarshalled timeVal, but I realize it always will.

Reply:
Ah, thank you for this note about inlining unmarshalling of timeVal. I didn't notice it was the same pattern but  you're right, it is!